Applications/Spire/Include/Spire/Ui/IconButton.hpp

```diff
@@ -22,48 +22,41 @@ namespace Spire {
       */
       explicit IconButton(QImage icon, QWidget* parent = nullptr);
 
-      //! Constructs an IconButton with a default icon and a hover icon.
+      //! Sets the icon to display.
       /*!
-        \param icon The icon shown when the button is not hovered.
-        \param hover_icon The icon shown when the button is hovered.
-        \param parent The parent QWidget to the IconButton.
+        \param icon The icon image.
       */
-      IconButton(QImage icon, QImage hover_icon, QWidget* parent = nullptr);
+      void set_icon(const QImage& icon);
```

Comment:
Why is this setter here instead of using the constructor?

Reply:
This is used in the case of the TitleBar, which is constructed with a default Spire icon but can later be replaced.

Reply:
If you want to replace it, then construct a new ```IconButton```.

Reply:
What's the benefit of replacing the button when all that's really required is to replace the icon?

Reply:
Fair question, the answer is to prefer a small API or at the very least, an API with as few setters as possible.